porcupine/plugins/filemanager.py

```diff
@@ -25,51 +25,84 @@
 else:
     trash_name = "trash"
 
+copy_path: Path | None = None
+
 
 def find_tabs_by_parent_path(path: Path) -> list[tabs.FileTab]:
     return [
         tab
         for tab in get_tab_manager().tabs()
         if isinstance(tab, tabs.FileTab)
-        and tab.path is not None
-        and (path == tab.path or path in tab.path.parents)
+           and tab.path is not None
+           and (path == tab.path or path in tab.path.parents)
     ]
 
 
-def ask_name_for_renaming(old_path: Path) -> Path | None:
+def ask_name_for_renaming(old_path: Path, is_paste=False) -> Path | None:
     label_width = 400
 
     dialog = tkinter.Toplevel()
     dialog.transient(get_main_window())
     dialog.resizable(False, False)
-    dialog.title("Rename")
 
     big_frame = ttk.Frame(dialog, padding=10)
     big_frame.pack(fill="both", expand=True)
-    ttk.Label(
-        big_frame, text=f"Enter a new name for {old_path.name}:", wraplength=label_width
-    ).pack(fill="x")
 
+    if is_paste and old_path.parent != copy_path.parent:
+        dialog.title("File conflict")
+        ttk.Label(
+            big_frame, text=f"{old_path.parent} already has a file named {old_path.name}."
+                            f"\nDo you want to overwrite it?",
+            wraplength=label_width
+        ).pack(fill="x")
+    else:
+        dialog.title("Rename")
+        ttk.Label(
+            big_frame, text=f"Enter a new name for {old_path.name}:", wraplength=label_width
+        ).pack(fill="x")
+
+    entry_frame = ttk.Frame(big_frame)
+    entry_frame.pack(fill="x")
     var = tkinter.StringVar()
-    entry = ttk.Entry(big_frame, textvariable=var)
-    entry.pack(pady=40, fill="x")
+    entry = ttk.Entry(entry_frame, textvariable=var)
+    entry.pack(pady=40, side=tkinter.BOTTOM, fill="x")
     entry.insert(0, old_path.name)
 
     button_frame = ttk.Frame(big_frame)
     button_frame.pack(fill="x", pady=(10, 0))
 
     new_path = None
+    overwrite_var = tkinter.BooleanVar(value=False)
 
     def select_name() -> None:
         nonlocal new_path
-        new_path = old_path.with_name(entry.get())
+        if not overwrite_var.get():
+            new_path = old_path.with_name(entry.get())
+        else:
+            new_path = old_path
         dialog.destroy()
 
     cancel_button = ttk.Button(button_frame, text="Cancel", command=dialog.destroy, width=1)
     cancel_button.pack(side="left", expand=True, fill="x", padx=(0, 5))
     ok_button = ttk.Button(button_frame, text="OK", command=select_name, state="disabled", width=1)
     ok_button.pack(side="right", expand=True, fill="x", padx=(5, 0))
 
+    def overwriting() -> None:
+        entry.config(state="disabled")
+        ok_button.config(state="enabled")
+
+    def renaming() -> None:
+        entry.config(state="enabled")
```

Comment:
I think the correct state is `"normal"`. Apparently non-existent states default to same as `"normal"`, so `"enabled"` works too, but `"normal"` would be more explicit.